### PR TITLE
Mailchimp accepts emails with apostrophes

### PR DESCRIPTION
**Pull request by gareth-lloyd**

### CODE REVIEW

**Comment by stephenross** [comment]
As a note for the confused, email addresses with the `'` character are apparently popular in countries like Ireland when people use their names in their email address. It is allowed per the RFC spec for email addresses. @gareth-lloyd, can you add an update to the docstring for that function and indicate the addition of the `'` to the regex so that people that look at the source of the original string don't try and send a PR 'correcting' it? Once that is done, we can merge this up.

File: mailchimp3/helpers.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def check_email(email):
     :type email: :py:class:`str`
     :return: Nothing
     """
-    if not re.match(r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)", email):
+    if not re.match(r"(^[a-zA-Z0-9_.+'-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)", email):
```
Comment (stephenross) [inline mailchimp3/helpers.py]: Please add information about the change to the docstring as mentioned in the pull request comments. Countries like Ireland that have apostrophes in their names use them in email and it is valid per the email spec. Once that is updated, we can merge this in.

**Comment by stephenross** [comment]
Closing this in favor of the voodoo in #99 as that also satisfies this

**Comment by stevejalim** [comment]
Hi Stephen - I have just hit this apostrophe issue, too, and was about to fork and offer up a patch. 

However, this PR does what I would have done. 

Any chance, given that #99 is in limbo, that this one could be merged after all? The author of #99 could rebase to include this small change if they wanted them

**Comment by stevejalim** [comment]
Looking at it again, I see this is in limbo too - is it worth me making a fresh PR with the relevant comments/documentation, too?

**Comment by stephenross** [comment]
@stevejalim If you would like to make a PR with the final changes from PR #99 and appropriate documentation as requested in that request, that would be fine as that person seems to be dragging their heels. It may be a good idea to simply fork their branch and perform the operations there so that they still retain some of the credit.

**Pull request closed without merging**